src/test/java/test/listeners/github1735/ExecutionListenerTest.java

```diff
@@ -0,0 +1,24 @@
+package test.listeners.github1735;
+
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import org.testng.xml.XmlSuite;
+import test.SimpleBaseTest;
+
+import java.util.Collections;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ExecutionListenerTest extends SimpleBaseTest {
+    @Test
+    public void ensureExecutionListenerIsInvokedOnlyOnce() {
+        XmlSuite suite = createXmlSuite("suite");
+        createXmlTest(suite, "test1", TestClassSample.class, TestClassTwoSample.class);
+        createXmlTest(suite, "test2", TestClassSample.class, TestClassTwoSample.class);
+        TestNG testng = create(suite);
+        testng.run();
+        assertThat(LocalExecutionListener.getFinish()).containsExactlyElementsOf(Collections.singletonList("finish"));
```

Comment:
Maybe `assertThat(LocalExecutionListener.getFinish()).containsExactly("finish");`

Reply:
I tried using this. But I think this would still pass, if the list contains multiple entries for `finish`. That is why I didn't use this.

Reply:
I can't remember which one but assertj provide one method which is exactly doing what you expect.